src/pcre2_jit_compile.c

```diff
@@ -11206,10 +11206,15 @@ cc += 1 + LINK_SIZE;
 
 if (opcode == OP_ONCE)
   {
+  int data;
+  int framesize = BACKTRACK_AS(bracket_backtrack)->u.framesize;
+
+  SLJIT_ASSERT(SHRT_MIN <= framesize && framesize < SHRT_MAX/2);
   /* We temporarily encode the needs_control_head in the lowest bit.
-     Note: on the target architectures of SLJIT the ((x << 1) >> 1) returns
```

Comment:
while there is documentation to explain why `x << 1` is done here as "temporary", it is not clear where this is used and neither why it is ok to override "framesize" this way.